analyses/cell-types-annotation/util/function-calculate-cell-type-signature.R

```diff
@@ -0,0 +1,120 @@
+  
+  for (i in 1:ncol(gene_markers_df)) {
+    cell.type <- paste(colnames(gene_markers_df)[i], ".score", sep = "")
+    gene.markers <- as.character(gene_markers_df[, i][gene_markers_df[, i] != ""])
+    
+    #if (genome_name == "GRCh38" | genome_name == "hg19"){
+     # gene.markers <- toupper(gene.markers)
+      
+    #} else if (genome_name == "mm10"){
+    #  gene.markers <- str_to_title(gene.markers)
+      
+    #} else if (genome_name == "Dualhg19"){
+   #  gene.markers <- paste("hg19-", toupper(gene.markers), sep = "")
+      
+    #} else if (genome_name == "DualGRCh38"){
+     # gene.markers <- paste("GRCh38-", toupper(gene.markers), sep = "")
+      
+    #} else if (genome_name == "Dualmm10"){
+    #  gene.markers <- paste("mm10---", str_to_title(gene.markers), sep = "") 
+      
+    #} else if (genome_name == "GRCm39") {
+    #  gene.markers <- paste("GRCm39---", str_to_title(gene.markers), sep = "") 
+      
+    #} else if (genome_name == "DualGRCm39") {
+     # gene.markers <- paste("DualGRCm39---", str_to_title(gene.markers), sep = "") } 
+
+    ####################################################
+    # TESTING PHASE ####################################
+    # sce <- as.SingleCellExperiment(seurat_obj)
+    # gene.markers <- gene.markers[gene.markers %in% getGenes(sce)]
+    test_rownames <- gene.markers[gene.markers %in% rownames(seurat_obj)] %>% as.data.frame()
+    write_tsv(test_rownames, file = paste0(results_dir, "/", "test_rownames", ".tsv")) 
+    
+    test_getGenes <- gene.markers[gene.markers %in% getGenes(seurat_obj)] %>% as.data.frame()
+    write_tsv(test_getGenes, file = paste0(results_dir, "/", "test_getGenes", ".tsv")) 
+    
+    # Check overlap (exact row matches)
+    overlap <- intersect(test_rownames, test_getGenes)
+    write_tsv(overlap, file = paste0(results_dir, "/", "test_overlap", ".tsv")) 
+    
+    # View the overlapping rows
+    # print(overlap)
+    
+    # Check if two data frames are exactly the same
+    is_identical <- all.equal(test_rownames, test_getGenes)
+    
+    # Print the result
+    if (is_identical == TRUE) {
+      print("The data frames overlap 100%!")
+    } else {
+      print("The data frames do not overlap 100%.")
+      print(is_identical)
+    }
+    
+    ####################################################    
+    
+    ####################################################    
+    gene.markers <- gene.markers[gene.markers %in% rownames(seurat_obj)] %>% as.data.frame()
+    # Save gene.markers used for the annotation
+    write_tsv(gene.markers, file = paste0(results_dir, "/", "gene.markers", ".tsv")) 
+  
+    seurat_obj <- AddModuleScore(seurat_obj, features = list(gene.markers), name = cell.type)
+    
+    # Print plots  
+    #print(FeaturePlot(seurat_obj, features = paste(cell.type, "1", sep=""), label = TRUE) + theme(aspect.ratio = 1))
+    print(VlnPlot(seurat_obj, features = paste(cell.type, "1", sep = ""), pt.size = 0))
```

Comment:
For this violin plot, I think it would be useful to have the option to specify which column to use for plotting the scores for each cluster.